### PR TITLE
Support of background transparency

### PR DESCRIPTION
**Pull request by Rhylx**
This change is [Reviewable](https://reviewable.io/reviews/qutebrowser/qutebrowser/4062)

### CODE REVIEW

File: qutebrowser/browser/webengine/webview.py [hunk 1]
```diff
@@ -173,7 +173,12 @@ def _set_bg_color(self):
         col = config.val.colors.webpage.bg
         if col is None:
             col = self._theme_color
-        self.setBackgroundColor(col)
+        if self.view() is not None:
+            if col.alpha() < 255:
+                self.view().setAttribute(Qt.WA_TranslucentBackground)
+            else:
+                self.view().resetAttribute(Qt.WA_TranslucentBackground)
+        self.setBackgroundColor(Qt.transparent)
```
Comment (The-Compiler) [inline qutebrowser/browser/webengine/webview.py]: This now always makes the color transparent, no matter what the user actually configured...

File: qutebrowser/browser/webkit/webview.py [hunk 2]
```diff
@@ -104,6 +104,12 @@ def _set_bg_color(self):
         palette = self.palette()
         if col is None:
             col = self.style().standardPalette().color(QPalette.Base)
+        if self.view() is not None :
+            if col.alpha() < 255 :
+                print(col.alpha())
```
Comment (The-Compiler) [inline qutebrowser/browser/webkit/webview.py]: This shouldn't be here

File: qutebrowser/mainwindow/mainwindow.py [hunk 3]
```diff
@@ -348,6 +357,9 @@ def _on_config_changed(self, option):
             self._update_overlay_geometries()
         elif option == 'window.hide_decoration':
             self._set_decoration(config.val.window.hide_decoration)
+        elif option == 'bg_transparency':
```
Comment (The-Compiler) [inline qutebrowser/mainwindow/mainwindow.py]: That option doesn't exist, you probably mean `colors.webpage.bg`

File: qutebrowser/browser/webengine/webview.py [hunk 4]
```diff
@@ -173,7 +173,12 @@ def _set_bg_color(self):
         col = config.val.colors.webpage.bg
         if col is None:
             col = self._theme_color
-        self.setBackgroundColor(col)
+        if self.view() is not None:
+            if col.alpha() < 255:
+                self.view().setAttribute(Qt.WA_TranslucentBackground)
+            else:
+                self.view().resetAttribute(Qt.WA_TranslucentBackground)
```
Comment (toofar) [inline qutebrowser/browser/webengine/webview.py]: >resetAttribute

I don't think this is a real thing. It certainly raised an AttributeError for me. It [looks](http://doc.qt.io/qt-5/qwidget.html#setAttribute) like you can use `setAttribute(..., False)`

**Comment by toofar** [comment]
And a robot has something to say about excessive whitespace: https://travis-ci.org/qutebrowser/qutebrowser/jobs/403018428

File: qutebrowser/browser/webengine/webview.py (hunk 1 above)
Comment (Rhylx) [reply qutebrowser/browser/webengine/webview.py]: Ok I modified it with : 
self.view().setAttribute(Qt.WA_TranslucentBackground, False)

File: qutebrowser/browser/webkit/webview.py (hunk 2 above)
Comment (Rhylx) [reply qutebrowser/browser/webkit/webview.py]: I don't understand why it shouldn't be here ?

File: qutebrowser/mainwindow/mainwindow.py (hunk 3 above)
Comment (Rhylx) [reply qutebrowser/mainwindow/mainwindow.py]: Ok I modified it

File: qutebrowser/browser/webengine/webview.py (hunk 4 above)
Comment (Rhylx) [reply qutebrowser/browser/webengine/webview.py]: I modified it, thanks for your help :)

File: qutebrowser/browser/webkit/webview.py (hunk 2 above)
Comment (The-Compiler) [reply qutebrowser/browser/webkit/webview.py]: Because you don't want qutebrowser to print out random numbers?

Comment (Rhylx) [reply qutebrowser/browser/webkit/webview.py]: Yes you're right. It's now modified. I read the code too fast.

File: qutebrowser/browser/webkit/webview.py [hunk 5]
```diff
@@ -104,6 +104,11 @@ def _set_bg_color(self):
         palette = self.palette()
         if col is None:
             col = self.style().standardPalette().color(QPalette.Base)
+        if self.view() is not None:
+            if col.alpha() < 255:
+                self.view().setAttribute(Qt.WA_TranslucentBackground)
+            else:
+                self.view().resetAttribute(Qt.WA_TranslucentBackground)
```
Comment (toofar) [inline qutebrowser/browser/webkit/webview.py]: I think this also needs to be changed to `setAttribute(..., False)`.

File: qutebrowser/browser/webengine/webview.py [hunk 6]
```diff
@@ -173,7 +173,12 @@ def _set_bg_color(self):
         col = config.val.colors.webpage.bg
         if col is None:
             col = self._theme_color
-        self.setBackgroundColor(col)
+        if self.view() is not None:
+            if col.alpha() < 255:
+                self.view().setAttribute(Qt.WA_TranslucentBackground)
+            else:
+                self.view().setAttribute(Qt.WA_TranslucentBackground, False)
+        self.setBackgroundColor(Qt.transparent)
```
Comment (toofar) [inline qutebrowser/browser/webengine/webview.py]: This can be changed back to `self.setBackgroundColor(col)` now.

**Comment by The-Compiler** [comment]
@Rhylx are you still interested in finishing this?

**Comment by The-Compiler** [comment]
Closing because of inactivity.

**Pull request closed without merging**